### PR TITLE
Add Database

### PR DESCRIPTION
**Pull request by End-rey**
I connect a database in project, implement a simple table 'Users' to test work of service

### CODE REVIEW

**Review by MicAPic — APPROVED** [review]
No objections. I suggest we focus on making an entity for projects and the related repository first -- only then shift our resources to the user repo.

File: pom.xml [hunk 1]
```diff
@@ -10,25 +10,34 @@
...
 	<properties>
 		<java.version>17</java.version>
 	</properties>
 	<dependencies>
-		<dependency>
+		<!-- <dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-security</artifactId>
-		</dependency>
+		</dependency> -->
```
Comment (SerjVankovich) [inline pom.xml:23]: Please, provide comment why did you comment this dependency

File: src/main/java/org/spburegistry/backend/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -1,46 +1,47 @@
...
-        return encoder;
-    }
-}
\ No newline at end of file
+//     @Bean
+//     public PasswordEncoder passwordEncoder() {
+//         PasswordEncoder encoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
+//         return encoder;
+//     }
+// }
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/config/SecurityConfiguration.java:47]: Add TODO to delete or modify this class in the future

File: src/main/java/org/spburegistry/backend/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+@RequestMapping("/data/")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    // @GetMapping("/users/{username}")
+    // public User getUserByName(@PathVariable String name) {
+    //     return userService.findByName(name);
+    // }
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/controller/UserController.java:21]: Why did you comment this method?

File: src/main/java/org/spburegistry/backend/entity/User.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "email")
+    private String email;
+
+    public User() {}
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/entity/User.java:25]: Why did you add empty constructor?

File: src/main/java/org/spburegistry/backend/entity/User.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+
+@Entity
+@Table(name="users")
+@Data
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/entity/User.java:13]: Please, add @Builder annotation

**Review by SerjVankovich — COMMENTED** [review]
Please, fix comments. BTW, everything is OK!

File: src/main/java/org/spburegistry/backend/controller/UserController.java (hunk 3 above)
Comment (End-rey) [reply src/main/java/org/spburegistry/backend/controller/UserController.java:21]: I think that i need this, but now i realize that no, i don't need this

File: src/main/java/org/spburegistry/backend/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("/users/user")
+    public ResponseEntity<User> getUser(@RequestParam(required = false) Long id) {
+        try {
+            User user = userService.findById(id);
+            return new ResponseEntity<>(user, HttpStatus.OK);
+        } catch (Exception e) {
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/controller/UserController.java:25]: Please, specify Exception type

File: src/main/java/org/spburegistry/backend/service/implementation/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    private final UserRepo userRepo;
+
+    @Autowired
+    public UserServiceImpl(UserRepo userRepo) {
+        this.userRepo = userRepo;
+    }
+
+    @Override
+    public User findById(long id) {
+        return userRepo.findById(id).get();
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/service/implementation/UserServiceImpl.java:21]: `Optional#get()` is not a good practice. Please, handle null value and return response with error.

Comment (SerjVankovich) [reply src/main/java/org/spburegistry/backend/service/implementation/UserServiceImpl.java:21]: I think, it is better to add ExceptionHandler to start work properly with these types of exception.

File: src/main/java/org/spburegistry/backend/ExceptionHandler/error/AppError.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package org.spburegistry.backend.ExceptionHandler.error;
+
+public class AppError {
+    private int statusCode;
+    private String message;
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/ExceptionHandler/error/AppError.java:5]: Use Lombok for constructors and getters/setters generation

File: src/main/java/org/spburegistry/backend/controller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("/users/user")
+    public User getUser(@RequestParam() Long id) {
+        return userService.findById(id);
+    }
+
+    @GetMapping("/users")
+    public ResponseEntity<Iterable<User>> getAllUsers() {
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/controller/UserController.java:26]: Avoid using ResponseEntity in controllers.

**Review by SerjVankovich — APPROVED** [review]
